src/libs/Middleware/Reauthentication.js

```diff
@@ -58,18 +58,14 @@ function Reauthentication(response, request, isFromSequentialQueue) {
                 // There are some API requests that should not be retried when there is an auth failure like
                 // creating and deleting logins. In those cases, they should handle the original response instead
                 // of the new response created by handleExpiredAuthToken.
-                const shouldRetry = lodashGet(request, 'data.shouldRetry');
+                // If the request was from the sequential queue we don't want to return, we want to run the reauthentication callback and retry the request
                 const apiRequestType = lodashGet(request, 'data.apiRequestType');
 
                 // For the SignInWithShortLivedAuthToken command, if the short token expires, the server returns a 407 error,
                 // and credentials are still empty at this time, which causes reauthenticate to throw an error (requireParameters),
                 // and the subsequent SaveResponseInOnyx also cannot be executed, so we need this parameter to skip the reauthentication logic.
                 const skipReauthentication = lodashGet(request, 'data.skipReauthentication');
-                if ((!shouldRetry && !apiRequestType) || skipReauthentication) {
-                    if (isFromSequentialQueue) {
-                        return data;
-                    }
-
+                if ((!isFromSequentialQueue && !apiRequestType) || skipReauthentication) {
```

Comment:
I updated this to:

```
if (!isFromSequentialQueue && (isDeprecatedRequest || skipReauthentication)) {
```

Because `SignInWithShortLivedAuthToken` explicitly passes `skipReauthentication` so we should skip it. And we should only consider skipping if it's not from the sequential queue. 

> // For the SignInWithShortLivedAuthToken command, if the short token expires, the server returns a 407 error,
    // and credentials are still empty at this time, which causes reauthenticate to throw an error (requireParameters),
    // and the subsequent SaveResponseInOnyx also cannot be executed, so we need this parameter to skip the reauthentication logic.


Lmk if this makes sense

Reply:
Yeah idk if you think that it's fine we can call a draw on it 😄  

IMO the `!isFromSequentialQueue` part makes this logic harder to parse and I'm not sure if it's really necessary. 

Another way to think about all this is that...

- Anything that is a deprecated request at this point is **not** going to be in the sequential queue (redundant condition)
- `skipReauthentication` is only used for a single command that doesn't enter the sequential queue (another redundant condition)
- All deprecated requests should have a `request.resolve()` method (unnecessary check to see if we have one)

So that leaves us with something like:

```js
// We don't want to reauthenticate in some cases so we will return the response as is
if (skipReauthentication) {
    return data;
}

// There should be not be any deprecated requests that require re-authentication at this point. 
// We will resolve the original promise with the response.
if (isDeprecatedRequest) {
    request.resolve(data);
    return;   
}

// Everything below this originated from the new API, does not use `Network.post()`, will always be retried, can't ever end up in the MainQueue.
```